cmd/serve/httplib/httplib.go

```diff
@@ -201,6 +204,18 @@ func NewServer(handler http.Handler, opt *Options) *Server {
 		}
 		oldHandler := handler
 		handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// No auth if the request is for the web-gui
+			fs.Debugf(nil, "URL:"+r.RequestURI)
+			match, err := regexp.MatchString(`(?m)^(/[^/]*)$`, r.RequestURI)
```

Comment:
This compiles the regexp for every request which is less efficient than it should be :-(

You could combine the two matches into a single regexp like this: https://play.golang.org/p/ZKGkNeaAnFs which is compiles once when the program starts.

Or you could write this without the regexp as

```go
   // URLs with an initial / and then lots of non slashes
    match = len(r.RequestURL) > 0 && r.RequestURI[0] == "/" && strings.IndexByte(r.RequestURI[1:], '/') == -1
```

I probably prefer the combined regexp approach as that looks more maintainable